src/components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as Styles from './style';
+
+class Tooltip extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.setTooltipPosition = this.setTooltipPosition.bind(this);
+    this.state = {
+      childWidth: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.setTooltipPosition();
+    window.addEventListener("resize", this.setTooltipPosition);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.setTooltipPosition);
+  }
+
+  setTooltipPosition() {
+    // Getting the first child width to calculate Tooltip position
+    const childWidth = this.tooltipWrapper.children[0].children[0].getBoundingClientRect().width;
+    this.setState({
+      childWidth,
+    });
+  }
+
+  /**
+   * Adjusting the styles according to the desired position
+   * The tooltip should be vertically or horizontally centered
+   */
+  getTooltipPosition(triggerRect, tooltipRect, position) {
+    const { childWidth } = this.state;
+    const gap = 8;
+    const triggerCenter = triggerRect.left + childWidth / 2;
+    const left = triggerCenter - tooltipRect.width / 2;
+    const maxLeft = window.innerWidth - tooltipRect.width - 2;
+    const verticalCenter = triggerRect.top + triggerRect.height - ((triggerRect.height + tooltipRect.height) / 2) - window.scrollY;
+
+    const newPosition = {
+      left: Math.min(Math.max(2, left), maxLeft) + window.scrollX,
+      top: triggerRect.bottom + gap + window.scrollY,
+    };
+
+    switch (position) {
+      case 'top': 
+        newPosition.top = triggerRect.top - tooltipRect.height - gap - window.scrollY;
+        break;
+
+      case 'right': 
+        newPosition.left = triggerRect.left + childWidth + gap + window.scrollX;
+        newPosition.top = verticalCenter;
+        break;
+
+      case 'left': 
+        newPosition.left = triggerRect.left - tooltipRect.width - gap - window.scrollX;
+        newPosition.top = verticalCenter;
+        break;
+
+      default:
+    }
+
+    return newPosition;
+  }
+
+  /**
+   * Rendering label with hotkey option if available
+   */
+  renderLabel = (label, hotkey) => (
+    <div>
+      {label && (
+        <Styles.Label
+          color='white'
+        >
+          <Styles.HotkeyWrapper>
+            {label}
+            {hotkey &&
+              <Styles.Label color="#B8B8B8" isHotkey>{hotkey}</Styles.Label>}
+          </Styles.HotkeyWrapper>
+        </Styles.Label>
+      )}
+    </div>
+  );
+  
+  render() {
+  const { children, label, position, hotkey } = this.props;
+  
+    return (
+      <Styles.TooltipWrapper innerRef={node => this.tooltipWrapper = node}>
+        <Styles.TooltipStyled
+          label={this.renderLabel(label, hotkey)}
+          position={(triggerRect, tooltipRect) => this.getTooltipPosition(triggerRect, tooltipRect, position)}
+        >
+          <div>
```

Comment:
@anafilipadealmeida one of the issues I'm getting is that for using the reach Tooltip, I need to add this div wrapping the children for it to work, which is not that big a deal, except, if I change it to be a `styled.div` it doesn't work, I get an error.

Reply:
Hey @MayaUribe, the error you get is because function components cannot be given refs, right? I would personally leave it like it is right now. However, I noticed we have styled-components package on "3.4.10", which still uses innerRefs. On version 4 they are deprecated in favor of the React 16 forwardRef. In a near future it might be worth considering upgrade it. Any thoughts on this @hamish?

Reply:
Hey there @anafilipadealmeida! That's correct, the error is due to stateless function components, they don't support refs, which causes the error in the reach Tooltip, since the ref is necessary to calculate positions and all that.
I think it might be a good idea to  upgrade `styled-components` package in the near future, but there might be some adjustments we'll have to make to do so 🤔

Reply:
Yeah, this is a good one @anafilipadealmeida and @MayaUribe! 

The version of `styled-components` is `3.*` in `@bufferapp/ui` because when we had it at `4.*` there were issues with buffer-analyze that @federicoweber wasn't able to easily resolve. That said **I'm pretty sure Federico has solved that issue by removing their snapshot testing.** And so, similar to the React version upgrade, I have a hunch we could move forward with upgrading styled-components as well. 

Perhaps one of you can reach out to the design systems team in Slack to just get a gut check on this and make sure it would work for other teams too? You could even publish a beta version of `@bufferapp/ui` with just styled-components upgraded so everyone can test.